client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -1069,6 +1077,39 @@ static boolean convertExistsResponse(Response response) {
         return response.getStatusLine().getStatusCode() == 200;
     }
 
+    /** Converts an eniter response into a json sting
+     *
+     * This is useful for responses that we don't parse on the client side, but instead work as string
+     * such as in case of the license JSON
+     */
+    static String convertResponseToJson(Response response) throws IOException {
```

Comment:
I was happy with the put license API using the license json because I figured most people don't edit their license. It is just a big blob of json they get from somewhere and stick into wherever it needs to go. I mean, I'd have been happy with using the `License` object from x-pack:plugin:core but it has so much stuff in it! There is like a file watcher in there. All kinds of stuff that has no business being in a client at all. I feel like it'd be nice to get fields like max_nodes and expiry back in the client but License.java will take a lot of work to move. So I kind of figured it wasn't worth doing. I kind of wonder how much license administration folks will do with the java client, but :shrug:.

Reply:
How about this? I can move `convertResponseToJson` into LicenseClient, rename `public String license()` into `public String getLicenseDefinition()` and we will leave it like this for now. If we realize at some point that we actually want License as License, we can always implement a client-side version of the License class and add `public License getLicense()` method that will return it. This way we don't really paint ourselves into a corner and make progress.

Reply:
I'm fine with that. Though I still think you might be ok with `EntityUtils.toString` so you don't really need the helper at all.